### DISCUSSION TITLE
Create GenericChart from Json

### DISCUSSION THREAD

**Discussion by uittorio** [plotly/Plotly.NET#421] [Q&A] [answered]
Hi, I am currently building an application where people can customise their template in json plotly standard (not sure if there is a standard, just the one used in javascript).

```json
{
  "data": [...],
   "layout": {...},
  "frames": [...]
}
```

Is it possible to  create a GenericGraph starting from the json without having to map each single property?

**Comment by kMutagene** [comment, marked as answer]
The thing with our strongly typed layers is that they unfortunately only work well for JSON _creation_. plotly.js JSON has properties that have arbitrary types, which you can do in a dynamically typed language such as JavaScript or python, but is quite hard to do in .NET.

We circumvent this on object creation by offering a strongly typed subset of possible types and by using `DynamicObj`, which lets us set members dynamically. However, this basically prevents deserialization - even if you could deserialize all nested objects into `DynamicObj` (which does not work), there is no way of knowing whether that `DynamicObj` represents a `Trace`, a `LinearAxis`, or any other strong type abstraction provided by Plotly.NET without looking at the name of the member and additional knowledge about the plotly JSON schema. 

Your use case is especially hard, because chart templates are basically allowing **all** possible plotly objects. I'd suggest only providing the options to set a subset of template properties and expand with what is needed. But maybe a pointer to the code where we implement the plotly template at least helps a little:

https://github.com/plotly/Plotly.NET/blob/11456b42a8875d3ddc6142e2f02129de229e5320/src/Plotly.NET/Templates/ChartTemplates.fs#L168-L701

**Comment by kMutagene** [comment]
Oh, quite hacky solution, but you could include the template as javascript in the output of this lib by patching the generated html string for a chart, e.g. by replacing the actual template in the output of `GenericChart.toChartHTML`

**Reply by uittorio** [reply]
Thank you for the explanation!

Another approach I might follow is to use directly the plotty js library and run the application through puppeter to generate the image!
